sql/src/main/java/org/apache/apex/malhar/sql/SQLExecEnvironment.java

```diff
@@ -0,0 +1,273 @@
+   *
+   * @return Returns this {@link SQLExecEnvironment}
+   */
+  public SQLExecEnvironment registerTable(String name, Endpoint endpoint)
+  {
+    Preconditions.checkArgument(name != null, "Table name cannot be null");
+    registerTable(name, new ApexSQLTable(schema, name, endpoint));
+    return this;
+  }
+
+  /**
+   * Register a table using {@link Table} with this {@link SQLExecEnvironment}
+   *
+   * @param name Name of the table that needs to be present with SQL statement
+   * @param table Object of type {@link Table}
+   *
+   * @return Returns this {@link SQLExecEnvironment}
+   */
+  public SQLExecEnvironment registerTable(String name, Table table)
+  {
+    Preconditions.checkArgument(name != null, "Table name cannot be null");
+    schema.add(name, table);
+    return this;
+  }
+
+  /**
+   * Register custom function with this {@link SQLExecEnvironment}
+   *
+   * @param name Name of the scalar SQL function that needs made available to SQL Statement
+   * @param fn Object of type {@link Function}
+   *
+   * @return Returns this {@link SQLExecEnvironment}
+   */
+  public SQLExecEnvironment registerFunction(String name, Function fn)
+  {
+    Preconditions.checkArgument(name != null, "Function name cannot be null");
+    schema.add(name, fn);
+    return this;
+  }
+
+  /**
+   * Register custom function from given static method with this {@link SQLExecEnvironment}
+   *
+   * @param name Name of the scalar SQL function that needs make available to SQL Statement
+   * @param clazz {@link Class} which contains given static method
+   * @param methodName Name of the method from given clazz
+   *
+   * @return Return this {@link SQLExecEnvironment}
+   */
+  public SQLExecEnvironment registerFunction(String name, Class clazz, String methodName)
+  {
+    Preconditions.checkArgument(name != null, "Function name cannot be null");
+    ScalarFunction scalarFunction = ScalarFunctionImpl.create(clazz, methodName);
+    return registerFunction(name, scalarFunction);
+  }
+
+  /**
+   * This is the main method takes SQL statement as input and contructs a DAG using contructs registered with this
+   * {@link SQLExecEnvironment}.
+   *
+   * @param sql SQL statement that should be converted to a DAG.
+   */
+  public void executeSQL(String sql)
+  {
+    FrameworkConfig config = buildFrameWorkConfig();
+    Planner planner = Frameworks.getPlanner(config);
+    try {
+      logger.info("Parsing SQL statement: {}", sql);
+      SqlNode parsedTree = planner.parse(sql);
+      SqlNode validatedTree = planner.validate(parsedTree);
+      RelNode relationalTree = planner.rel(validatedTree).rel;
+      logger.info("RelNode relationalTree generate from SQL statement is:\n {}",
+          Util.toLinux(RelOptUtil.toString(relationalTree)));
+      RelNodeVisitor visitor = new RelNodeVisitor(dag, typeFactory);
+      visitor.traverse(relationalTree);
+    } catch (Exception e) {
+      throw new RuntimeException(e);
```

Comment:
The difference is that `throw new RuntimeException(e)` can wrap another `RuntimeException`.